### DISCUSSION TITLE
Turning off Accessibility for Ghostty Crashes Mac

### DISCUSSION THREAD

**Discussion by tuladhar** [ghostty-org/ghostty#3936] [Issue Triage]
## Description
When the Accessibility permission for Ghostty is disabled, the Mac crashes immediately. The issue seems to be linked to the Accessibility settings for Ghostty, as disabling this feature causes the system to freeze.

### Steps to Reproduce
1. Install and launch Ghostty on macOS.
2. Go to System Settings > Privacy & Security > Accessibility
3. Find Ghostty under the Accessibility section and enable the permission.
4. Turn off the Accessibility permission for Ghostty.
5. Observe the Mac crashing upon turning off the permission.

## Expected Behavior
Turning off the Accessibility permission for Ghostty should not cause the Mac to crash. The app should either remain functional or prompt the user with an appropriate error message.

## Actual Behavior
When the Accessibility permission is turned off, the Mac crashes.

## System Information
- macOS Version: Sonoma 14.5
- Ghostty Version: 1.0.0 (build: 8469, commit: 4b4d4062)
- Hardware: Apple M1 Pro

**Comment by edrozenberg** [comment]
Mac mini M4 Sequoia 15.2 (24C101).

No crash, but toggling the perm multiple times and triggering a global shortcut did cause the Mac to temporarily freeze, was eventually able to quit Ghostty and everything seems back to normal.

Note that standard Mac behavior is to only ask for perms once, probably to prevent "harassment" from repeated perms requests. If you enable then disable the Accessibility perms subsequently there won't be any further prompts, the user would need to remember to toggle the perms on again to make global things work, the OS will not remind again.

If the Mac crashes that would be an OS issue, opening a ticket in Apple Developer feedback will ensure that nobody from Apple ever responds, but maybe when enough people complain Apple would pay attention after a decade :). Many of my dozens of Apple issues are still present many years later.

**Reply by frangio** [reply]
I can confirm this. Reproduction steps:

1. Add a global binding like the one suggested for the quick terminal: `keybind = global:cmd+backquote=toggle_quick_terminal`
2. Reload the config. If this is the first global keybind Ghostty will request Accessibility permissions, grant them.
3. Remove Accessibility permissions in macOS Settings.
4. macOS should seem frozen now.
5. I was able to get out of this by locking then unlocking, focusing the Settings window and re-enabling Accessibility permissions.

To avoid this, any global keybinds should first be removed from the config, Ghostty config reloaded, and only then can Accessibility be removed.

**Reply by panoptic** [reply]
Didn't have to lock or log out. Mouse functionality returned after a while, maybe ~10 seconds*. Then navigated to the accessibility settings via the Apple menu, re-enabled Ghostty -> system was fine.

\* After the duration, the keyboard accepted single keypresses, which however froze both mouse and keyboard again.

Tahoe 26.2.

**Comment by bo2themax** [comment]
I believe this is already fixed long time ago. Please open another one if you think otherwise
